### DISCUSSION TITLE
CI Feature: Implement pull request preview environments

### DISCUSSION THREAD

**Discussion by waveywaves** [databendlabs/databend#9295] [Ideas]
**Summary**

I would like to support Databend by implementing [Uffizzi](https://github.com/UffizziCloud/uffizzi) preview environments.
Disclaimer: I work on [Uffizzi](https://github.com/UffizziCloud/uffizzi).
Uffizzi is a Open Source full stack previews engine and our platform is available completely free for Databend  (and all open source projects). This will provide maintainers with preview environments of every PR in the cloud, which enables faster iterations and reduces time to merge. You can see the open source repos which are currently using Uffizzi over [here](https://uffizzi.notion.site/)
Uffizzi is purpose-built for the task of previewing PRs and it integrates with your workflow to deploy preview environments in the background without any manual steps for maintainers or contributors.      We can go ahead and create an Initial PoC for you right away if you think there is value in this proposal.

**Comment by Xuanwo** [comment]
Hi @waveywaves, [uffizzi](https://github.com/UffizziCloud/uffizzi) seems cool! And thanks for reaching us, it's a honor to be chosen from numerous open source projects.

I have the following questions, please correct me if I understand incorrectly:

- Does implement `uffizzi` support mean we can test/preview it as `vercel` does for the frontend website?
- The only case in databend to take manual action to merge PR is a benchmark test. Does `uffizzi` suitable for this use-case?
- Databend is a database project which only depends on itself, for my first glance, it's not the target user of `uffizzi`. What do you think?

**Reply by waveywaves** [reply]
> Does implement uffizzi support mean we can test/preview it as vercel does for the frontend website?

exactly. we support all kinds of applications and architectures, including CLI, databases etc.

> The only case in databend to take manual action to merge PR is a benchmark test. Does uffizzi suitable for this use-case?

Does this mean you would like to run a benchmark test against an uffizzi preview ?

> Databend is a database project which only depends on itself, for my first glance, it's not the target user of uffizzi. What do you think?

We do support just databases. If you want to instantiate it with some test data we can do that as well. Is there a CLI or a UI with the database which you would like to use to interact with the database ?

**Reply by Xuanwo** [reply]
> Does this mean you would like to run a benchmark test against an uffizzi preview ?

Yep, this is the use case I can image on uffizzi. Do you have other ideas on what databend can benefit from a preview environment?

> Is there a CLI or a UI with the database which you would like to use to interact with the database ?

all client tools of mysql/clickouse/pg, and of course [bendsql](https://github.com/databendcloud/bendsql)

**Reply by waveywaves** [reply]
@Xuanwo I would like to ask you the same question :) What is the ideal preview environment you see for databend and we can try to provide that for you or at least something close
